pxe-server/build/Dockerfile

```diff
@@ -3,15 +3,73 @@
 
 FROM debian:12.11-slim AS builder
 
+# Set shell with pipefail for safer pipe operations
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
         dnsmasq=2.90-4~deb12u1 && \
     rm -rf /var/lib/apt/lists/*
 
+# Download sources for dependent packages and create tar archive
+RUN mkdir -p /sources && \
+    apt-get update && \
+    # Install dpkg-dev for source package handling
+    apt-get install -y --no-install-recommends dpkg-dev=1.21.22 tar=1.34+dfsg-1.2+deb12u1 xz-utils=5.4.1-1 && \
+    # Create source repositories
+    echo "deb-src http://deb.debian.org/debian bookworm main" > /etc/apt/sources.list.d/debian-sources.list && \
+    echo "deb-src http://deb.debian.org/debian-security bookworm-security main" >> /etc/apt/sources.list.d/debian-sources.list && \
+    # Update with --allow-insecure-repositories to avoid signing conflicts
+    apt-get update --allow-insecure-repositories || apt-get update && \
+    # Define package list with versions
+    echo "dnsmasq:2.90-4~deb12u1" > /tmp/target_packages.txt && \
+    echo "dnsmasq-base:2.90-4~deb12u1" >> /tmp/target_packages.txt && \
+    echo "libdbus-1-3:1.14.10-1~deb12u1" >> /tmp/target_packages.txt && \
```

Comment:
These package versions hardcoded if dnsmasq get updated with new version then we have update these dependencies version right? Please add a new if that is the case.